agent/recorder.go

```diff
@@ -149,7 +157,18 @@ func (r *SpanRecorder) SendSpans() (error, bool) {
 		}
 		statusCode, err := r.callIngest(buf)
 		if err != nil {
-			if statusCode == 401 {
+			if statusCode == 0 {
```

Comment:
Let's try to use https://github.com/hashicorp/go-retryablehttp. What we are doing here is not specific to us, so I'd try to use that library which I think should have enough configuration options to meet our use case.

Reply:
I have to take a look to this lib, I need to keep the current counters needed to print the report and keep the behavior for the 401 response.

Also I saw that we need to stay in an old version to support Go 1.11

> Version 0.6.0 and before are compatible with Go prior to 1.12. From 0.6.1 onward, Go 1.12+ is required.

Reply:
It is configurable so we could have made it work for our use case, but I agree that the lack of support for go 1.11 is a problem.